src/pages/UsersPage.tsx

```diff
@@ -1,14 +1,39 @@
 import { useEffect, useState } from "react";
-import { getRegisteredUsers, RegisteredUser } from "../api/services/users";
+import { ImUserPlus } from "react-icons/im";
+import {
+  getRegisteredUsers,
+  inviteUser,
+  RegisteredUser
+} from "../api/services/users";
+import { Modal } from "../components";
+import {
+  InviteUserForm,
+  InviteUserFormValue
+} from "../components/InviteUserForm";
 import { SearchLayout } from "../components/Layout";
-import { toastError } from "../components/Toast/toast";
+import { toastError, toastSuccess } from "../components/Toast/toast";
 import { UserList } from "../components/UserList";
 import { useLoader } from "../hooks";
 
 export function UsersPage() {
   const [users, setUsers] = useState<RegisteredUser[]>([]);
+  const [isOpen, setIsOpen] = useState(false);
   const { loading, setLoading } = useLoader();
 
+  const onSubmit = async (val: InviteUserFormValue) => {
+    try {
+      const { error } = await inviteUser(val);
+      if (!error) {
```

Comment:
Just a little observation here, are we ignoring the error here on purpose? If that's not the case, we could re-throw it so that it can be caught by the catch, as shown below.

```ts
if (error) {
  throw error;
}

const userName = `${val.firstName} ${val.lastName}`;
toastSuccess(`${userName} invited successfully`);
setIsOpen(false);
fetchUsersList();
```